tools/keytools/sign.py

```diff
@@ -123,13 +123,20 @@ def make_header(image_file, fw_version, extra_fields=[]):
     for t in extra_fields:
         tag = t[0]
         sz = t[1]
+        if sz == 4:
+            while (len(header) % 4) != 0:
+                header += struct.pack('B', HDR_PADDING)
+        elif sz == 8:
+            while (len(header) % 8) != 4:
+                header += struct.pack('B', HDR_PADDING)
```

Comment:
does not seem to do the same thing. 

When the value is 4B long, align the header to the beginning of the word:
XX XX XX XX    XX XX XX FF   [Hdr 4 bytes]   [Value, 4B aligned]

When the value is 8B long, the header should be aligned to 8 + 4. e.g.:

XX FF FF FF   [Hdr 4 bytes]   [Value, 8 byte aligned]

A valid shorter alternative may be:
```
if sz in [4, 8]:
    while (len(header) % sz != (sz - 4)):
          header += struct.pack('B', HDR_PADDING)
```
But is far less readable and I still prefer the original change.

Reply:
Now I understand that the field needs to be 4 or 8 byte aligned. These require this code point aligning to after writing the 4 bytes of tag and length, that it be 4 or 8 byte aligned. If we refactor this code, I'd really like to see it generalized into a method and align any type to type length and account for the tag and length. I think it would clean up the code and make it so we can test it independently.

```
HDR_TAG_LEN=4
            n_stuff = (sz - len(header) - HDR_TAG_LEN) % sz
            header += n_stuff * struct.pack('B', HDR_PADDING)
```